From 0a4882b9b69238b140c83d619f4d6bb6a2c8a146 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:39 -0400
Subject: [tools] perf: Use hpp formats to sort hist entries

Message-id: <1410181752-25631-215-git-send-email-jolsa@redhat.com>
Patchwork-id: 91872
O-Subject: [PATCH RHEL7.1 BZ1134356 214/307] perf tools: Use hpp formats to sort hist entries
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 093f0ef34c50ff5cca41c1e18e258ff688e915b6
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Mar 3 12:07:47 2014 +0900

description
===========
It wrapped sort entries to hpp functions, so using the hpp sort list
to sort entries.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/hist.c b/tools/perf/util/hist.c
index 7f0236c..38373c9 100644
--- a/tools/perf/util/hist.c
+++ b/tools/perf/util/hist.c
@@ -432,11 +432,11 @@ struct hist_entry *__hists__add_entry(struct hists *hists,
 int64_t
 hist_entry__cmp(struct hist_entry *left, struct hist_entry *right)
 {
- struct sort_entry *se;
+ struct perf_hpp_fmt *fmt;
  int64_t cmp = 0;
 
- list_for_each_entry(se, &hist_entry__sort_list, list) {
-  cmp = se->se_cmp(left, right);
+ perf_hpp__for_each_sort_list(fmt) {
+  cmp = fmt->cmp(left, right);
   if (cmp)
    break;
  }
@@ -447,15 +447,11 @@ hist_entry__cmp(struct hist_entry *left, struct hist_entry *right)
 int64_t
 hist_entry__collapse(struct hist_entry *left, struct hist_entry *right)
 {
- struct sort_entry *se;
+ struct perf_hpp_fmt *fmt;
  int64_t cmp = 0;
 
- list_for_each_entry(se, &hist_entry__sort_list, list) {
-  int64_t (*f)(struct hist_entry *, struct hist_entry *);
-
-  f = se->se_collapse ?: se->se_cmp;
-
-  cmp = f(left, right);
+ perf_hpp__for_each_sort_list(fmt) {
+  cmp = fmt->collapse(left, right);
   if (cmp)
    break;
  }
-- 
1.7.1